### ISSUE TITLE
Add ignoreFontFamilies: [] to font-family-no-missing-generic-family-keyword

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#3288]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

The `stylelint-config-recommended` shared config includes the `font-family-no-missing-generic-family-keyword` rule ([src](https://github.com/stylelint/stylelint-config-recommended/blob/master/index.js#L17))

The issue I have is that I'm using an [icon font](https://css-tricks.com/examples/IconFont/) and as there is (typically) no _fallback_ for icon fonts so `font-family: icons;` will trigger the `font-family-no-missing-generic-family-keyword` rule.

> Which rule, if any, is this issue related to?

`font-family-no-missing-generic-family-keyword`

> What CSS is needed to reproduce this issue?

```css
[data-icon]:before {
  font-family: icons; /* BYO icon font, mapped smartly */
  content: attr(data-icon);
  speak: none; /* Not to be trusted, but hey. */
}
```

> What stylelint configuration is needed to reproduce this issue?

Use or extend the `stylelint-config-recommended` shared config

> Which version of stylelint are you using?

`9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

n/a

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No
> What did you expect to happen?

Warnings happen as expected 

> What actually happened (e.g. what warnings or errors you are getting)?

n/a

I can envisage a couple of options to resolve this:

1. Remove the `font-family-no-missing-generic-family-keyword` rule from `stylelint-config-recommended`

2. Override the `font-family-no-missing-generic-family-keyword` rule from my config

3. Add an `ignore` option to the `font-family-no-missing-generic-family-keyword` rule where a font name can be declared e.g. `icons` and the `font-family-no-missing-generic-family-keyword` rule would then ignore these instances.

I've a preference for option 3 which I'm thinking would be the best option, thoughts @stylelint/core ?

**Comment by evilebottnawi** [comment]
@ntwb In some projects we use very difference font's name for icons. Default ignore very dirty solution. Maybe better add option `ignoreSeleletor` (or better name) where ignore this

**Comment by puggan** [comment]
Maybe there should be a new generic-family in the css-standard, like "hidden", that hide the text if the font isn't loaded. Then we can use `font-family: icons, hidden;`. But I have no ide how to sugest changes to the css-standard.

**Comment by jeddy3** [comment]
I'm still in favour of using `/* stylelint-disable */` commands in these instances, for the reasons outlined in https://github.com/stylelint/stylelint/issues/3048#issuecomment-349409778. Mainly that font icons are a workaround that is going away.

Having said that, this is the second request for such an option. So, I'm not opposed to adding an `ignoreValues: []` option. I believe it would cover everyone's use cases e.g.

```json
"font-family-no-missing-generic-family-keyword": [ true, {
  "ignoreValues": [
    "icons",
    "evilebottnawi-icon-font-1",
    "evilebottnawi-icon-font-2"
  ]
}
```

The recommended config would remain unchanged for the reasons above, though.

**Comment by YozhikM** [comment]
What is the final decision on the rule?

**Comment by hudochenkov** [comment]
@YozhikM [add new option](https://github.com/stylelint/stylelint/issues/3288#issuecomment-389649183) `ignoreValues`.

**Comment by simbawus** [comment]
@jeddy3  the "ignoreValues" option didn't work

**Comment by jeddy3** [comment]
@simbawus This issue is labelled as "help wanted".  Please consider contributing this option. There's [a section in the Developer Guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-an-option-to-an-existing-rule) on how to get started. 

There's a PR, https://github.com/stylelint/stylelint/pull/3736, that adds a similar option to another rule - you can use that as a blueprint for this option.

**Comment by glen-84** [comment]
@puggan,

> But I have no ide how to sugest changes to the css-standard.

https://lists.w3.org/Archives/Public/www-style/ (probably)

**Comment by rgant** [comment]
A work around for this is to use a variable to contain the font-family:

```scss
$icon-font: 'Material Icons';

button {
    font-family: $icon-font;
}
```

**Comment by dr0bz** [comment]
> A work around for this is to use a variable to contain the font-family:
> 
> ```css-scss
> $icon-font: 'Material Icons';
> 
> button {
>     font-family: $icon-font;
> }
> ```

Doing the same here:)

**Comment by jeddy3** [comment]
Please consider [contributing](https://stylelint.io/contributing) this option if anyone has time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.
